paimon-core/src/main/java/org/apache/paimon/jdbc/JdbcUtils.java

```diff
@@ -0,0 +1,646 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.jdbc;
+
+import org.apache.paimon.catalog.Identifier;
+
+import org.apache.paimon.shade.guava30.com.google.common.collect.Maps;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/** Util for jdbc catalog. */
+public class JdbcUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcUtils.class);
+    public static final String METADATA_LOCATION_PROP = "metadata_location";
```

Comment:
If I understand correctly, Paimon considers the commit successful once the snapshot file is successfully committed to the file system.  Storing the current metadata location in the catalog metastore will cause consistency problems because the two operations of committing the snapshot file to the file system and committing it to the metastore are not atomic. So there seems to be no point in storing this metadata location？ @JingsongLi Can you please review this? Thanks.

Reply:
@zhongyujiang You understanding is very correct! Yes, Paimon only consider `Snapshot` files as source of truth.
`metadata_location` is come from Iceberg, we don't need this.